src/EVM/SymExec.hs

```diff
@@ -567,14 +567,14 @@ verify solvers opts preState maybepost = do
       Nothing -> pure (expr, [Qed ()])
       Just post -> liftIO $ do
         let
-          -- Filter out any leaves that can be statically shown to be safe
-          canViolate = flip filter flattened $
-            \leaf -> case Expr.simplifyProp (post preState leaf) of
-              PBool True -> False
-              _ -> True
+          -- Filter out any leaves from `flattened` that can be statically shown to be safe
+          tocheck = flip map flattened $ \leaf -> (toPropsFinal leaf, leaf)
+            where
+              toProps leaf = PNeg (post preState leaf) : assumes <> extractProps leaf
+              toPropsFinal leaf = if opts.simp then Expr.simplifyProps $ toProps leaf
+                                               else toProps leaf
```

Comment:
Can we put this in the existing `where` block? I find multiple sets of where blocks kind hard to parse

Reply:
Good point! I fixed it now :)